### DISCUSSION TITLE
Guide for adding a new machine

### DISCUSSION THREAD

**Discussion by fire-lizard** [86Box/86Box#4692] [General]
I would like to create a pull request for adding a new machine (**IBM 5271** aka **3270 PC**) . I have got all the ROMs, but while I have seen existing adding machines pull requests and got decent C++ skill (at least I hope so) I don't fully understand the pattern for writing code.
Could anyone please give me some guidelines if possible?
(Of course, I presume that maybe there is no pattern at all)

**Comment by jriwanek** [comment]
So, you want to look at the machine table for the 5270, (IBM AT), you'll also want to add an init function prototype in machine.h, and clone the init for the 5270 for it in m_at.c (Editing accordingly), new devices go in src/device in their own.c file, take a look at ibm_5161.c for about the closest thing for a template for devices,

I'd advise joining the discord to interact with us.
